microservices/download/utils.js

```diff
@@ -1,4 +1,44 @@
-const superagent = require("superagent");
+const https = require("node:https")
+
+function doRequest(queryString) {
+  const dataToSend = JSON.stringify({ query: queryString })
+
+  const options = {
+    hostname: 'api.cirrus-ci.com',
+    path: '/graphql',
+    method: 'POST',
+    headers: {
+      'Accept': 'application/json',
+      'Content-Type': 'application/json',
+      'Content-Length': dataToSend.length
+    }
+  }
+
+  return new Promise((resolve, reject) => {
+    let data = '';
+
+    const req = https.request(options, (res) => {
+      res.setEncoding('utf8');
+      res.on('data', (chunk) => {
+        data += chunk;
+      });
+      res.on('end', () => {
+        // No more data in response.
+        resolve(JSON.parse(data))
+      });
+    });
+
+    req.on('error', (e) => {
+      console.error(`problem with request: ${e.message}`);
+      throw e
+      reject(e)
+    });
```

Comment:
This is the error handling in the new HTTPS POST function I added.

If you prefer not to throw an error here, this may be important to change.

I chose to throw here mostly for ease of debugging -- If we don't throw here, the error message is less helpful, since it gives no filename or line number that the error comes from, IIRC.

Reply:
Yeah, okay, I'm pretty sure I need to not throw here.

I didn't really pay attention to the existing error handling, since I was going to re-use this file for a different purpose where I didn't need to handle error situations gracefully.

---

Unrelated, I could move this new function I wrote `doRequest()` inside of the existing `findLink()` that uses it, so it's more obvious they're related and only ever used together. Makes the file flow a little easier to skim/read, too.

Reply:
Update: I pushed a new commit that makes it so this no-longer throws for errors during the HTTPS POST request, just bubbles the error up the stack with `reject()` so the rest of the file's error handling can deal with it, essentially as it was before.

Testing showing the errors are virtually identical before/after this PR now, with the change to not `throw`.

Before, on `main`:

```console
% node test-find-link.js
Error: getaddrinfo ENOTFOUND api.cirrus-ci.comx
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:109:26) {
  errno: -3008,
  code: 'ENOTFOUND',
  syscall: 'getaddrinfo',
  hostname: 'api.cirrus-ci.comx',
  response: undefined
}
{ ok: false, code: 505, msg: 'Server Error' }
```

This PR:

```console
% node test-find-link.js
Error: getaddrinfo ENOTFOUND api.cirrus-ci.comx
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:109:26) {
  errno: -3008,
  code: 'ENOTFOUND',
  syscall: 'getaddrinfo',
  hostname: 'api.cirrus-ci.comx'
}
{ ok: false, code: 505, msg: 'Server Error' }
```

Only difference is 'superagent' added a "response: `undefined`" field to their error output.